### PR TITLE
Adds multi-GPU training script for tiny, small, and medium heterogeneous IGB dataset

### PR DESCRIPTION
**Pull request by Elnifio**
[Related Issue](https://github.com/IllinoisGraphBenchmark/IGB-Datasets/issues/27). In this PR, we add a training script for IGB heterogeneous dataset. This training script is verified on medium IGB-Heterogeneous dataset.

### CODE REVIEW

**Comment by Elnifio** [comment]
The script is modified based on [train_multi_gpu.py](https://github.com/IllinoisGraphBenchmark/IGB-Datasets/blob/main/igb/train_multi_gpu.py). In `line 215, 216`, we additionally adds the system arguments to control the schedulers' hyperparameters (step size and gamma).

File: igb/train_multi_hetero.py
Comment (Elnifio) [inline igb/train_multi_hetero.py:226]: At this line, we are utilizing the `IGBHeteroDGLDataset` instead of `IGBHeteroDGLDatasetMassive`, same as the implementation in [`train_hetero.py`](https://github.com/IllinoisGraphBenchmark/IGB-Datasets/blob/28ca6b343d75c170e515e62d5e4f9f03d5553f77/igb/train_hetero.py#L182), so this script is not verified for large and full dataset.

**Comment by msharmavikram** [comment]
Thanks for the contributions @Elnifio!

**Comment by akhatua2** [comment]
Thanks for adding this script @Elnifio. Overall LGTM! Please dry run (1 epoch on tiny) it once and comment the output before you merge it.

**Review by akhatua2 — CHANGES_REQUESTED** [review]
Thanks for adding this script @Elnifio. Overall LGTM! Please dry run (1 epoch on tiny) it once and comment the output for a sanity check.

**Comment by msharmavikram** [comment]
@Elnifio  Could you also update README.md file for the changes on how to run `train_multi_hetero.py`  I expect this to be similar to 

```
python train_multi_hetero.py --model_type rgcn --dataset_size tiny --num_classes 19 --gpu_devices 0,1,2,3
```

**Comment by msharmavikram** [comment]
Seems like it works. Accuracy is almost the same. 

For single GPU:
```
python train_multi_hetero.py --model_type rgcn --dataset_size tiny --num_classes 19 --gpu_devices 0 --path ./
Graph(num_nodes={'author': 357041, 'fos': 84220, 'institute': 8738, 'paper': 100000},
      num_edges={('author', 'affiliated_to', 'institute'): 325410, ('paper', 'cites', 'paper'): 547416, ('paper', 'topic', 'fos'): 718445, ('paper', 'written_by', 'author'): 471443},
      metagraph=[('author', 'institute', 'affiliated_to'), ('paper', 'paper', 'cites'), ('paper', 'fos', 'topic'), ('paper', 'author', 'written_by')])
model size: 3.271MB
Epoch 000 | Loss 13.2862 | Train Acc 34.00 | Val Acc 55.02 | Time 0:00:09 | GPU 1300.5 MB
Epoch 005 | Loss 5.8810 | Train Acc 67.81 | Val Acc 68.91 | Time 0:00:09 | GPU 1596.1 MB
Epoch 010 | Loss 5.5008 | Train Acc 69.05 | Val Acc 69.49 | Time 0:00:09 | GPU 1599.1 MB
Epoch 015 | Loss 5.3866 | Train Acc 69.19 | Val Acc 69.81 | Time 0:00:09 | GPU 1603.9 MB
Test Acc 69.69%
Total time taken 0:02:02
```

For multiple GPU:
```
python train_multi_hetero.py --model_type rgcn --dataset_size tiny --num_classes 19 --gpu_devices 0,1 --path ./
Graph(num_nodes={'author': 357041, 'fos': 84220, 'institute': 8738, 'paper': 100000},
      num_edges={('author', 'affiliated_to', 'institute'): 325410, ('paper', 'cites', 'paper'): 547416, ('paper', 'topic', 'fos'): 718445, ('paper', 'written_by', 'author'): 471443},
      metagraph=[('author', 'institute', 'affiliated_to'), ('paper', 'paper', 'cites'), ('paper', 'fos', 'topic'), ('paper', 'author', 'written_by')])
model size: 3.271MB
Epoch 000 | Loss 7.8814 | Train Acc 25.31 | Val Acc 40.79 | Time 0:00:08 | GPU 1299.1 MB
Epoch 005 | Loss 3.5840 | Train Acc 62.95 | Val Acc 65.03 | Time 0:00:09 | GPU 1597.5 MB
Epoch 010 | Loss 2.9587 | Train Acc 67.52 | Val Acc 69.06 | Time 0:00:09 | GPU 1597.5 MB
Epoch 015 | Loss 2.8033 | Train Acc 68.80 | Val Acc 69.43 | Time 0:00:09 | GPU 1603.7 MB
Test Acc 69.21%
Total time taken 0:02:11
```

**Review by akhatua2 — APPROVED** [review]
Merging to main.

**Comment by Elnifio** [comment]
@msharmavikram Thanks for providing the logs! I was blocked on other topics earlier today and didn't have time to run it. Glad to see that this PR is merged :) Our previous test shows that on medium IGBH dataset, the accuracy under _default argument_ is slightly worse (~1% lower) than 1-GPU. 

As for the instructions on how to run this script, it is the same as running `train_multi_gpu.py`, with added `--sched_stepsize` and `--sched_gamma` on lines `215, 216` to control the scheduler's hyperparameter, so I previously did not modify the README for the execution instruction.

**Comment by msharmavikram** [comment]
No issues. We updated the readme. :)

**Pull request merged**
